### PR TITLE
add pv padding and hf model handling to backtest_sites.py

### PR DESCRIPTION
**Pull request by AUdaltsova**
# Pull Request

## Description

- Added an option to use a model from HuggingFace
- Added datapipe to pad pv data so that sun position is created correctly ([see this issue in dpipes](https://github.com/openclimatefix/ocf-data-sampler/issues/67))
- Fixed conversion to nanoseconds warning

## How Has This Been Tested?

Ran small backtest sthat require pv padding with a local and a HF model

- [X] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

Checked that with new coded padding the forecast is identical to what a manually padded dataset produces.

- [X] Yes

## Checklist:

- [X] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: scripts/backtest_sites.py [hunk 1]
```diff
@@ -67,6 +70,10 @@
 # checkpoint on the val set
 model_chckpoint_dir = "PLACEHOLDER"
 
+revision = None
```
Comment (Sukh-P) [inline scripts/backtest_sites.py:73]: A small one but maybe these could have `hf_` at the beginning just so it's immediately clear they are hf related variables?

File: scripts/backtest_sites.py [hunk 2]
```diff
@@ -412,7 +477,13 @@ def main(config: DictConfig):
     # Create a dataloader for the concurrent batches and use multiprocessing
     dataloader = DataLoader(batch_pipe, **dataloader_kwargs)
     # Load the PVNet model
-    model, *_ = get_model_from_checkpoints([model_chckpoint_dir], val_best=True)
+    if model_chckpoint_dir is not None:
```
Comment (Sukh-P) [inline scripts/backtest_sites.py:480]: I think this could be simplified to 
```suggestion
    if model_chckpoint_dir:
```
and similarly the `elif` below

File: scripts/backtest_sites.py [hunk 3]
```diff
@@ -412,7 +477,13 @@ def main(config: DictConfig):
...
+        model, *_ = get_model_from_checkpoints([model_chckpoint_dir], val_best=True)
+    elif model_id is not None:
```
Comment (AUdaltsova) [inline scripts/backtest_sites.py:482]:
```suggestion
    elif model_id:
```

**Review by Sukh-P — APPROVED** [review]
Brilliant work adding these fixes/features in!

File: scripts/backtest_sites.py (hunk 1 above)
Comment (AUdaltsova) [inline scripts/backtest_sites.py:73]:
```suggestion
hf_revision = None
```

File: scripts/backtest_sites.py [hunk 4]
```diff
@@ -67,6 +70,10 @@
 # checkpoint on the val set
 model_chckpoint_dir = "PLACEHOLDER"
 
+revision = None
+token = None
+model_id = None
```
Comment (AUdaltsova) [inline scripts/backtest_sites.py:75]:
```suggestion
hf_token = None
hf_model_id = None
```

File: scripts/backtest_sites.py [hunk 5]
```diff
@@ -412,7 +477,13 @@ def main(config: DictConfig):
     # Create a dataloader for the concurrent batches and use multiprocessing
     dataloader = DataLoader(batch_pipe, **dataloader_kwargs)
     # Load the PVNet model
-    model, *_ = get_model_from_checkpoints([model_chckpoint_dir], val_best=True)
+    if model_chckpoint_dir:
+        model, *_ = get_model_from_checkpoints([model_chckpoint_dir], val_best=True)
+    elif model_id:
+        model = load_model_from_hf(model_id, revision, token)
```
Comment (AUdaltsova) [inline scripts/backtest_sites.py:483]:
```suggestion
    elif hf_model_id:
        model = load_model_from_hf(hf_model_id, hf_revision, hf_token)
```

File: scripts/backtest_sites.py (hunk 1 above)
Comment (AUdaltsova) [reply scripts/backtest_sites.py]: yes good idea!

File: pyproject.toml [hunk 6]
```diff
@@ -14,7 +14,7 @@ dependencies = [
     "xarray",
     "ipykernel",
     "h5netcdf",
-    "torch>=2.0.0",
+    "torch>=2.0.0, <2.5.0",
```
Comment (AUdaltsova) [inline pyproject.toml:17]:
```suggestion
    "torch>=2.0.0",
```

File: scripts/backtest_sites.py [hunk 7]
```diff
@@ -101,11 +108,64 @@
 # FUNCTIONS
 
 
+@functional_datapipe("pad_forward_pv")
+class PadForwardPVIterDataPipe(IterDataPipe):
+    """
+    Pads forecast pv. Sun position is calculated based off of pv time index
```
Comment (AUdaltsova) [inline scripts/backtest_sites.py:114]:
```suggestion
    Pads forecast pv. 
    
    Sun position is calculated based off of pv time index
```

File: scripts/backtest_sites.py [hunk 8]
```diff
@@ -101,11 +108,64 @@
...
+
+        for xr_data in self.pv_dp:
+            t0 = xr_data.time_utc.data[int(xr_data.attrs["t0_idx"])]
+            pv_step = np.timedelta64(xr_data.attrs["sample_period_duration"])
+            t_end = t0 + self.forecast_duration + pv_step
+            time_idx = np.arange(xr_data.time_utc.data[0], t_end, pv_step)
+            yield xr_data.reindex(time_utc=time_idx, fill_value=-1)
+
+
+def load_model_from_hf(model_id: str, revision: str, token: str):
```
Comment (AUdaltsova) [inline scripts/backtest_sites.py:137]:
```suggestion
def load_model_from_hf(model_id: str, revision: str, token: str):

"""
Loads model from HuggingFace
"""
```

File: scripts/backtest_sites.py [hunk 9]
```diff
@@ -101,11 +108,71 @@
...
+
+"""
+Loads model from HuggingFace
+"""
```
Comment (AUdaltsova) [inline scripts/backtest_sites.py:143]:
```suggestion
    """
    Loads model from HuggingFace
    """
```
